libbpf-cargo/src/btf/btf.rs

```diff
@@ -993,6 +1038,49 @@ impl Btf {
                         writeln!(def, r#"}}"#)?;
                     }
                 }
+                BtfType::Enum64(t) => {
```

Comment:
This is fine in principle, but please let's start introducing functions for newly added branches:
```patch
--- libbpf-cargo/src/btf/btf.rs
+++ libbpf-cargo/src/btf/btf.rs
@@ -765,6 +765,52 @@ impl Btf {
         }
     }

+
+    fn enum64_def(btf: &BtfEnum64<'_>, def: &mut String) -> Result<()> {
+        let repr_size = match btf.size {
+            1 => "8",
+            2 => "16",
+            4 => "32",
+            8 => "64",
+            _ => bail!("Invalid enum64 size: {}", btf.size),
+        };
+
+        let signed = if btf.signed { "i" } else { "u" };
+
+        writeln!(def, r#"#[derive(Debug, Copy, Clone, PartialEq, Eq)]"#)?;
+        writeln!(def, r#"#[repr({signed}{repr_size})]"#)?;
+        writeln!(def, r#"pub enum {name} {{"#, name = btf.name)?;
+
+        for value in &btf.values {
+            writeln!(
+                def,
+                r#"    {name} = {value},"#,
+                name = value.name,
+                value = value.value,
+            )?;
+        }
+
+        writeln!(def, "}}")?;
+
+        // write an impl Default for this enum
+        if !btf.values.is_empty() {
+            // TODO: remove #[allow(clippy::derivable_impls)]
+            //       once minimum rust at 1.62+
+            writeln!(def, r#"#[allow(clippy::derivable_impls)]"#)?;
+            writeln!(def, r#"impl Default for {name} {{"#, name = btf.name)?;
+            writeln!(def, r#"    fn default() -> Self {{"#)?;
+            writeln!(
+                def,
+                r#"        {name}::{value}"#,
+                name = btf.name,
+                value = btf.values[0].name
+            )?;
+            writeln!(def, r#"    }}"#)?;
+            writeln!(def, r#"}}"#)?;
+        }
+        Ok(())
+    }
+
     /// Returns rust type definition of `ty` in string format, including dependent types.
     ///
     /// `ty` must be a struct, union, enum, or datasec type.
@@ -1038,49 +1084,7 @@ impl Btf {
                         writeln!(def, r#"}}"#)?;
                     }
                 }
-                BtfType::Enum64(t) => {
-                    let repr_size = match t.size {
-                        1 => "8",
-                        2 => "16",
-                        4 => "32",
-                        8 => "64",
-                        _ => bail!("Invalid enum64 size: {}", t.size),
-                    };
-
-                    let signed = if t.signed { "i" } else { "u" };
-
-                    writeln!(def, r#"#[derive(Debug, Copy, Clone, PartialEq, Eq)]"#)?;
-                    writeln!(def, r#"#[repr({signed}{repr_size})]"#)?;
-                    writeln!(def, r#"pub enum {name} {{"#, name = t.name)?;
-
-                    for value in &t.values {
-                        writeln!(
-                            def,
-                            r#"    {name} = {value},"#,
-                            name = value.name,
-                            value = value.value,
-                        )?;
-                    }
-
-                    writeln!(def, "}}")?;
-
-                    // write an impl Default for this enum
-                    if !t.values.is_empty() {
-                        // TODO: remove #[allow(clippy::derivable_impls)]
-                        //       once minimum rust at 1.62+
-                        writeln!(def, r#"#[allow(clippy::derivable_impls)]"#)?;
-                        writeln!(def, r#"impl Default for {name} {{"#, name = t.name)?;
-                        writeln!(def, r#"    fn default() -> Self {{"#)?;
-                        writeln!(
-                            def,
-                            r#"        {name}::{value}"#,
-                            name = t.name,
-                            value = t.values[0].name
-                        )?;
-                        writeln!(def, r#"    }}"#)?;
-                        writeln!(def, r#"}}"#)?;
-                    }
-                }
+                BtfType::Enum64(t) => Self::enum64_def(t, &mut def)?,
                 BtfType::Datasec(t) => {
                     let mut sec_name = t.name.to_string();
                     if sec_name.is_empty() || !sec_name.starts_with('.') {
```